lib/Bagger/Storage/Config.pm

```diff
@@ -0,0 +1,118 @@
+=head1 NAME
+
+   Bagger::Storage::Config -- Configuration Routines for Bagger Storage
+
+=cut
+
+package Bagger::Storage::Config;
+
+=head1 SYNOPSIS
+
+   my $config = Bagger::Storage::Config(
+             key   => 'additional_index_types', 
+	     value => ['rum', 'sp-gist']
+   );
+   $config->save
+
+=cut
+
+use strict;
+use warnings;
+use Carp 'croak';
+use Moose;
+use PGObject::Util::DBMethod;
+use PGObject::Type::JSON;
+use namespace::autoclean;
+use Scalar::Util 'reftype';
+use Moose::Util::TypeConstraints;
+with 'Bagger::Storage::PGObject';
+PGObject::Type::JSON->register; # become the codec for Perl <-> json
+                                # but not jsonb by default
+
+=head1 DESCRIPTION
+
+ The Bagger Storage Config module exists to configure anything needed regarding
+ Bagger storage nodes as a cluster.  It is expected that this data will somehow
+ be synchronized to the storage nodes themselves along with dimensions and
+ index information.
+
+ The configuration system is intended to be flexible and portable.  Values can
+ be anything, but undef is not accepted and a value must be provided.  Strings,
+ array and hashrefs, and references are all supported.
+
+ The data values are serialized to JSON in the database, which has one important
+ side effect.  If you pass a scalar reference to value, then when you retrieve
+ that reference, it will be returned as a scalar.
+
+=head1 PROPERTIES
+
+=over
+
+=item id Int, assigned by the database
+
+=cut
+
+has id => (is => 'ro', isa => 'Int');
+
+=item key Str, required
+
+=cut
+
+has key => (is => 'ro', isa => 'Str', required => 1);
+
+=item value -- Arbitrary scalar, hashref, or arrayref.  Required
+
+=item value_string -- For scalars, handle as string rather json conversion ref
+ 
+  use this one if you expect a string or number back.
+
+=cut
+
+subtype 'PGObject::Type::JSON'
+  => as 'PGObject::Type::JSON';
+
+coerce 'PGObject::Type::JSON'
+  => from 'Str | Ref'
+  =>  via { croak if (defined reftype($_)) and (reftype($_) =~ /CODE|SCALAR/);
+           return PGObject::Type::JSON->new($_) };
+
+
+has value => (is      => 'ro', 
+             isa      => 'PGObject::Type::JSON',
+             required => 1,
+	     coerce   => 1,
+             );
+sub value_string {
```

Comment:
I'm not quite getting this syntax. This is a sub, but it's supposed to be a property. It's probably a moose intricacy I'm not getting, but I'd like to understand.

Reply:
It's basically a custom accessor which is why I am listing it as a property.

Internally value is also a subroutine that wraps an internal field.

Reply:
(this just has an extra piece of logic to that retrieval)

Reply:
I could do this as a lazy attribute too with a custom builder.   Maybe that would be clearer?

Reply:
Ok changed this to a handle with an inserted method.

Reply:
Though I am thinking it might actually be better to subclass and add methods for our use.  I will do that in a separate PR.